cdmtaskservice/condor/client.py

```diff
@@ -0,0 +1,145 @@
+"""
+A client for submitting jobs to a remote HTCondor instance targeted for use by the CDM
+Task Service.
+"""
+
+import asyncio
+import htcondor2
+from pathlib import Path
+import posixpath
+from yarl import URL
+
+from cdmtaskservice.arg_checkers import not_falsy as _not_falsy, require_string as _require_string
+from cdmtaskservice import models
+
+
+# A lot of this is copied from
+# https://github.com/kbase/execution_engine2/blob/develop/lib/execution_engine2/utils/Condor.py
+STATIC_SUB = {
+    #  Allow up to 12 hours of no response from job
+    "JobLeaseDuration": "43200",
+    #  Allow up to 12 hours for condor drain
+    "MaxJobRetirementTime": "43200",
+    # Hold jobs running longer than 7 days
+    "Periodic_Hold": "( RemoteWallClockTime > 604800 )",
+    # If a job exits incorrectly put it on hold
+    "on_exit_hold": "ExitCode =!= 0",
+    "should_transfer_files": "yes",
+    "when_to_transfer_output": "on_exit_or_evict",
+    # For some reason setting transfer_output_files to the empty string isn't working.
+    # We specify an empty file so that large job output doesn't get transferred.
+    "transfer_output_files": "__DUMMY_OUTPUT__",
+    "getenv": "False"
+}
+
+
+class CondorClient:
+    """
+    The condor client.
+    """
+
+    def __init__(
+        self,
+        schedd: htcondor2.Schedd,
+        initial_dir: Path,
+        executable_url: str,
+        runner_archive_url: str,
+        client_group: str | None = None
+    ):
+        """
+        Create the client.
+        
+        schedd: An htcondor Schedd instance, configured to submit jobs to the cluster.
+        initial_dir - where the job should start on the condor worker. The path must exist there.
```

Comment:
https://batchdocs.web.cern.ch/tutorial/exercise4a.html the initial dir is for submission and output files not for where it starts on the worker.